src/Components/Publishing/Sections/FullscreenViewer/__tests__/FullScreenProvider.test.js

```diff
@@ -0,0 +1,69 @@
+import { mount } from "enzyme"
+import React from "react"
+import renderer from "react-test-renderer"
+
+import { FullScreenProvider } from "../FullScreenProvider"
+
+describe("FullScreenProvider", () => {
+  it("injects props via the render props pattern", done => {
+    mount(
+      <FullScreenProvider>
+        {props => {
+          expect(Object.keys(props).sort()).toEqual([
+            "closeViewer",
+            "onViewFullscreen",
+            "openViewer",
+            "slideIndex",
+            "viewerIsOpen",
+          ])
+          done()
```

Comment:
Nice, this test helps me understand the pattern.

Reply:
Yeah `this.props.children` can be exectued as a function and parameters passed: https://github.com/artsy/reaction/pull/557/files#diff-f1d7f446bf9c5c77dd1446b1b5851b01R63